python/python/glide/async_commands/cluster_commands.py

```diff
@@ -742,3 +744,72 @@ async def random_key(self, route: Optional[Route] = None) -> Optional[str]:
             Optional[str],
             await self._execute_command(RequestType.RandomKey, [], route),
         )
+
+    async def scan(
+        self,
+        cursor: ClusterScanCursor,
+        match: Optional[str] = None,
+        count: Optional[int] = None,
+        type: Optional[ObjectType] = None,
+    ) -> List[Union[ClusterScanCursor, List[str]]]:
+        """
+        Incrementally iterates over the keys in the Cluster.
+        The method returns a list containing the next cursor and a list of keys.
+
+        This command is similar to the SCAN command, but it is designed to work in a Cluster environment.
+        For each iteration the new cursor object should be used to continue the scan.
+        Using the same cursor object for multiple iterations will result in the same keys or unexpected behavior.
+        For more information about the Cluster Scan implementation,
+        see [Cluster Scan](https://github.com/aws/glide-for-redis/wiki/General-Concepts#cluster-scan).
+
+        As the SCAN command, the method can be used to iterate over the keys in the database,
+        to return all keys the database have from the time the scan started till the scan ends.
+        The same key can be returned in multiple scans iteration.
+
+        See https://valkey.io/commands/scan/ for more details.
+
+        Args:
+            cursor (ClusterScanCursor): The cursor object that wraps the scan state.
+              To start a new scan, create a new empty ClusterScanCursor using ClusterScanCursor().
+            match (Optional[str]): A pattern to match keys against.
+            count (Optional[int]): The number of keys to return in a single iteration.
+              The actual number returned can vary and is not guaranteed to match this count exactly.
+              This parameter serves as a hint to the server on the number of steps to perform in each iteration.
+              The default value is 10.
+            type (Optional[ObjectType]): The type of object to scan for (STRING, LIST, SET, ZSET, HASH).
```

Comment:
I think we're better of removing the types list here - it;s just another place of maintaining this list when it changes